src/.vitepress/mermaid/MermaidDiagram.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <div ref="containerRef" class="mermaid-diagram" v-html="content"></div>
+</template>
+
+<script lang="ts">
+const createLoadMermaid = () => {
+  let mermaid: Promise<typeof import("mermaid")> | null = null;
+  return () => {
+    if (!mermaid) {
+      mermaid = import("mermaid").then((m) => {
+        m.default.initialize({
+          startOnLoad: false,
+          theme: "default",
+        });
+        return m;
+      });
+    }
+    return mermaid;
+  };
+};
+
+// Mermaidが使われていないページでMermaidを読み込むのを防ぐために、
+// Mermaidを別のバンドルに隔離しておいて、必要になるまで読み込まないようにする。
+const loadMermaid = createLoadMermaid();
```

Comment:
コメントは上に書いたほうが良さそう感

`const createLoadMermaid = () => {`の弱点なんですが、これ関数定義を上に書かないとなんですよね。
functionだったら下に書けるのに。

例えばこうするのどうでしょう。

```suggestion
// Mermaidが使われていないページでMermaidを読み込むのを防ぐために、
// Mermaidを別のバンドルに隔離しておいて、必要になるまで読み込まないようにする。
const loadMermaid = createLoadMermaid();

function createLoadMermaid {
  let mermaid: Promise<typeof import("mermaid")> | null = null;
  return () => {
    if (!mermaid) {
      mermaid = import("mermaid").then((m) => {
        m.default.initialize({
          startOnLoad: false,
          theme: "default",
        });
        return m;
      });
    }
    return mermaid;
  };
}
```

ついでに`const hoge = () => {`は微妙派になるのはどうでしょう。